api/pom.xml

```diff
@@ -135,12 +135,27 @@
         <plugins>
             <plugin>
                 <artifactId>maven-compiler-plugin</artifactId>
-                <version>3.1</version>
+                <version>3.8.1</version>
                 <configuration>
-                    <source>1.8</source>
-                    <target>1.8</target>
-                    <compilerArgument>-Xlint:all</compilerArgument>
+                    <release>9</release>
+                    <compilerArgs>
+                        <arg>-Xlint:all</arg>
+                    </compilerArgs>
                 </configuration>
+                <executions>
+                    <execution>
+                        <id>base-compile</id>
+                        <goals>
+                            <goal>compile</goal>
+                        </goals>
+                        <configuration>
+                            <release>8</release>
```

Comment:
I don't think we need or should build with java8.   That will only result in different builds for the same version of the api.

Reply:
I thought all Jakarta EE9 spec artifacts were supposed to be Java 8 bytecode.

Reply:
I just checked the master branch of jsp-api, jaxb-api, and cdi specs.
They all target `<release>1.8</release>` on their compile.

Reply:
So is EE9 is java8 byte code.... and this PR excludes module-info.java if built with java8, then this is a noop?  Or am I miss reading this?

Reply:
There were 2 `maven-compiler-plugin` entries in this PR when it was first submitted.

1. First one did a normal bytecode compile on release 8 without `module-info.java`.
2. Next one did a compile that only included `module-info.java` using release 9.

The later `package` step bundled them up into a single jar.

Reply:
I haven't changed the build configuration since the first time I've submitted.
It's actually pretty close to what is done in [JPA api build](https://github.com/eclipse-ee4j/jpa-api/blob/master/api/pom.xml#L254)